### ISSUE TITLE
Inconsistent print between typescript and flow parser

### ISSUE THREAD

**Issue by fisker** [prettier/prettier#14457]
**Prettier 2.8.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAggRmwF5sB6AKmwBZtzTsAKAHSm2wB88W3OAhFgJQBtALogANCAgYYAS2gBnZKACGAJzUQA7gAV1CJShUAbLSrRLJAIzUqwAazgwAyioC2cADKyocZADMTBThrWwcnZww7HwBzZBg1AFcQkDg3KzgAE0yszxUoGMSVGLgAMQg1NxUYOQLkEBVEmAgJEAALGDdjAHU22XgFKLA4ZwN+2QA3frR6sAVLEB9gtRgdWxiqgKCUgCsFAA9nWOM4AEVEiHgt42DJKLVl+vQsBTA1WRlWjHfYbtlMmBtZAADgADHdNMFurYMPVvnBlhM-JIAI4XeBraSGBoKAC0viyWVaajgaNkJLWxU2SECNxSwTcsniSXpxzguBq7ysTTgOjgam8vmutxACjZ50ufhp20kMBUVj+AKBSAATLLbLJjLEAMIQNzU1IKACsrUSwQAKvLDLSRRNkgBJKA5WDON4fGC4J3OdAnYVwAC+-qAA)

```sh
--parser typescript
```

**Input:**

```tsx
type A1 = /* 4 */ (
  | A
  | B
)[]
```

**Output:**

```tsx
type A1 = /* 4 */ (A | B)[];

```

**Prettier 2.8.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAggRmwF5sB6AKmwBZtzTsAKAHSm2wB88W3OAhFgJQBtALogANCAgYYAS2gBnZKACGAJzUQA7gAV1CJShUAbLSrRLJAIzUqwAazgwAyioC2cADKyocZADMTBThrWwcnZww7HwBzZBg1AFcQkDg3KzgAE0yszxUoGMSVGLgAMQg1NxUYOQLkEBVEmAgJEAALGDdjAHU22XgFKLA4ZwN+2QA3frR6sAVLEB9gtRgdWxiqgKCUgCsFAA9nWOM4AEVEiHgt42DJKLVl+v9jbVaMNR8YbtlMmDbkAAcAAY7ppgt1bBh6u84MsJn5JABHC7wNbSQwNBQAWl8WSyrTUcGRskJa2KmyQgRuKWCblk8SSNOOcFwNQ+ViacB0cDU3l811uIAUzPOlz8lO2khgKis31+-yQACYpbZZMZYgBhCBuCmpBQAVlaiWCABUZYYqYKJskAJJQHKwZxgD4yXD25zoE4CuAAXx9QA)

```sh
--parser flow
```

**Input:**

```jsx
type A1 = /* 4 */ (
  | A
  | B
)[]
```

**Output:**

```jsx
type A1 /* 4 */ = (A | B)[];

```

**Expected behavior:**

Should have same result.

**Comment by fisker** [comment]
https://github.com/facebook/flow/issues/9016

**Comment by fisker** [comment]
Fixed by #14555
